clients/include/testing_local_vector.hpp

```diff
@@ -112,4 +112,1280 @@ void testing_local_vector_bad_args(void)
+    T max = static_cast<T>(1.0);
+    vec.SetRandomUniform(min, max);
+
+    // Validate that all elements are within the specified range
+    for(int i = 0; i < vec.GetSize(); ++i)
+    {
+        EXPECT_GE(vec[i], min);
+        EXPECT_LE(vec[i], max);
+    }
+}
+
+template <typename T>
+void testing_set_random_normal()
+{
+    // Create a LocalVector and allocate some size
+    LocalVector<T> vec = getTestVector<T>(1000);
+
+    // Set random values in the vector
+    T mean = static_cast<T>(0.0);
+    T std  = static_cast<T>(1.0);
+    vec.SetRandomNormal(mean, std);
+
+    // Validate that the mean of all elements is close to the specified mean
+    T sum = 0.0;
+    for(int i = 0; i < vec.GetSize(); ++i)
+    {
+        sum += vec[i];
+    }
+    T calculated_mean = sum / vec.GetSize();
+    EXPECT_NEAR(calculated_mean, mean, 3 * std);
+
+    // Validate that the standard deviation of all elements is close to the specified std
+    T sum_sq = 0.0;
+    for(int i = 0; i < vec.GetSize(); ++i)
+    {
+        sum_sq += (vec[i] - calculated_mean) * (vec[i] - calculated_mean);
+    }
+    T calculated_std = std::sqrt(sum_sq / vec.GetSize());
+    EXPECT_NEAR(calculated_std, std, 3 * std);
+}
+
+template <typename T>
+void testing_copy_from()
+{
+    // Create a LocalVector and allocate some size
+    LocalVector<T> vec = getTestVector<T>(5);
+
+    // Create another LocalVector to copy from
+    LocalVector<T> src_vec = getTestVector<T>(5);
+
+    // Copy values from the source vector
+    EXPECT_NO_THROW(vec.CopyFrom(src_vec));
+
+    // Validate the copied values
+    for(int i = 0; i < vec.GetSize(); ++i)
+    {
+        EXPECT_EQ(vec[i], src_vec[i]);
+    }
+}
+
+template <typename T>
+void testing_clone_from()
+{
+    // Create a LocalVector and allocate some size
+    LocalVector<T> vec = getTestVector<T>(5);
+
+    // Clone the vector
+    LocalVector<T> cloned_vec;
+    EXPECT_NO_THROW(cloned_vec.CloneFrom(vec));
+
+    // Validate the cloned values
+    for(int i = 0; i < cloned_vec.GetSize(); ++i)
+    {
+        EXPECT_EQ(cloned_vec[i], vec[i]);
+    }
+}
+
+template <typename T>
+void testing_binary()
+{
+    // Create a LocalVector and allocate some size
+    LocalVector<T> vec = getTestVector<T>(5);
+
+    // Define the filename for the test
+    std::string filename = "test_vector.bin";
+
+    // Write the vector to a binary file
+    EXPECT_NO_THROW(vec.WriteFileBinary(filename));
+
+    // Read the vector from the binary file
+    LocalVector<T> read_vec;
+    EXPECT_NO_THROW(read_vec.ReadFileBinary(filename));
+
+    // Validate the read values
+    for(int i = 0; i < read_vec.GetSize(); ++i)
+    {
+        EXPECT_EQ(read_vec[i], vec[i]);
+    }
+
+    // Clean up the temporary file
+    std::remove(filename.c_str());
+}
+
+template <typename T>
+void testing_add_scale()
+{
+    // Create a LocalVector and allocate some size
+    LocalVector<T> v = getTestVector<T>(5);
+    LocalVector<T> x = getTestVector<T>(5);
```

Comment:
Why only length 5 vectors? Can't we also pass the `Arguments` struct for the `LocalVector` tests so we can specify the length of vectors we want to test in the test definition?